src/include/execution/compiler/operator/insert_translator.h

```diff
@@ -1 +1,128 @@
-#pragma once
\ No newline at end of file
+#pragma once
+
+#include "execution/compiler/operator/operator_translator.h"
+#include "planner/plannodes/insert_plan_node.h"
+
+#include <execution/compiler/storage/pr_filler.h>
+
+namespace terrier::execution::compiler {
+
+/**
+ * Insert Translator
+ */
+class InsertTranslator : public OperatorTranslator {
+ public:
+
+  InsertTranslator(const terrier::planner::InsertPlanNode *o, CodeGen *codegen);
+
+  /// Declare the inserter
+  void InitializeStateFields(util::RegionVector<ast::FieldDecl *> *state_fields) override;
```

Comment:
The inserter is local to one pipeline. It should be a local variable rather than part of the global state.